hibernate-core/src/main/java/org/hibernate/dialect/AbstractTransactSQLDialect.java

```diff
@@ -43,7 +43,7 @@ public AbstractTransactSQLDialect() {
 		registerColumnType( Types.BOOLEAN, "bit" );
 
 		//SQL Server/Sybase 'bit' type is always exactly one bit
-		registerColumnType( Types.BIT, "smallint" );
+		registerColumnType( Types.BIT, "bit" );
```

Comment:
This change is incorrect, which is why there is a comment there. The `bit` type in these databases is a literal bit and can't be null. That's why it's set to `smallint`.

Reply:
http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36271.1550/html/blocks/X34468.htm

Also they can't have indexes which is a problem if you want to put 'em in a PK.

Reply:
Ok, I just looked into the SQL Server documentation and there it states null is possible: https://docs.microsoft.com/en-US/sql/t-sql/data-types/bit-transact-sql?view=sql-server-ver15

Reply:
Does it make sense to use `bit` then for boolean if it doesn't support null on Sybase?

Reply:
Wait, I'm confused. How do we have:

    registerColumnType( Types.BOOLEAN, "bit" );

I thought that was wrong.

Reply:
I right, I see. Sorry, my explanation for that line is wrong. What's wrong with mapping `Types.BIT` to Transact SQL `bit` is that on Transact SQL the `bit` type is always exactly one bit. 

But `Dialect` already has a registration for that case:

    registerColumnType( Types.BIT, 1, "bit" );

This line:

    registerColumnType( Types.BIT, "smallint" );

is just overriding the mapping for length>1. Honestly I'm not sure if it's even worth having it there and perhaps we should simply delete that line. But whatever, it certainly doesn't make sense to map length>1 to `bit` in Transact SQL.

(It's `SybaseASEDialect` that handles the issue with mapping `Types.BOOLEAN` to Transact SQL `bit`, since it seems I misremembered and that is an issue only on ASE, not on SQL Server or Sybase Anywhere.)

Reply:
Two lines above. That was already there, but I guess this might have been a mistake then.

Reply:
So there's actually 5 different `Dialect`s which map `Types.BIT` to `smallint` for length>1. This is something we inherited from H5 it seems, and it's inconsistent (H2 doesn't do it, for example), and honestly it's not clear to me that it even makes all that much sense.

So I dunno, delete it or leave it be, I don't much care. The main danger is that it's confusing to people maintaining the `Dialect`s.